test/unit/database/database.spec.ts

```diff
@@ -409,5 +406,101 @@ describe('Database', () => {
         return db.setRules(rules).should.eventually.be.rejectedWith('network error');
       });
     });
+
+    describe('emulator mode', () => {
+      interface EmulatorTestConfig {
+        name: string;
+        setUp: () => FirebaseApp;
+        tearDown?: () => void;
+        url: string;
+      }
+
+      const configs: EmulatorTestConfig[] = [
```

Comment:
Optional: Consider merging these tests with the tests above. You can likely re-use EmulatorTestConfig (once renamed) and also validate the production URLs.

Reply:
This is a pretty good idea. However, upon trying it, required quite a bit of changes to the existing tests. Let's revisit this again in a future PR.